File: data/code_pairs/fortran/1586.f90
! This Fortran program demonstrates a simple module with a function and a 
! program that tests this function.

MODULE LmdVmixMod
IMPLICIT NONE
CONTAINS
    FUNCTION DoSomething(x) RESULT(y)
        REAL, INTENT(IN) :: x
        REAL :: y
        y = x * 2.0
    END FUNCTION DoSomething
END MODULE LmdVmixMod

PROGRAM TestLmdVmixMod
    USE LmdVmixMod
    IMPLICIT NONE
    REAL :: result

    result = DoSomething(5.0)
    PRINT *, "Result of DoSomething(5.0): ", result
END PROGRAM TestLmdVmixMod

File: data/code_pairs/cpp/1586.cpp
#include <iostream>

// Declaration of the LmdVmixMod class
class LmdVmixMod {
public:
    // Method that mimics the Fortran function
    double doSomething(double x) {
        return x * 2.0;
    }
};

int main() {
    LmdVmixMod obj; // Instantiate LmdVmixMod
    double result = obj.doSomething(5.0); // Use the method

    // Output the result
    std::cout << "Result of doSomething(5.0): " << result << std::endl;

    return 0;
}